broadlink/climate.py

```diff
@@ -201,3 +201,294 @@ def set_schedule(self, weekday: List[dict], weekend: List[dict]) -> None:
+            data['speed'] = 'mid'
+        elif speed_L == 0x20 and speed_R == 0x00:
+            data['speed'] = 'high'
+        elif speed_L == 0x40 and speed_R == 0x80:
+            data['speed'] = 'mute'
+        elif speed_R == 0x40:
+            data['speed'] = 'turbo'
+        elif speed_L == 0xa0 and speed_R == 0x00:
+            data['speed'] = 'auto'
+        else:
+            data['speed'] = 'unrecognized value'
+
+        data['sleep'] = False if (payload[0x11] & 0b100 == 0b000) else True
+        data['display'] = (payload[0x16] & 0x10 == 0x10)
+        data['health'] = (payload[0x14] & 0b11 == 0b11)
+        data['cmnd_0d_rmask'] = payload[0x0d] & 0xf
+        data['cmnd_0e_rmask'] = payload[0x0e] & 0xf
+        data['cmnd_18'] = payload[0x18]
+
+        checksum = self._calculate_checksum(payload[:0x19]) # checksum=(payload[0x1a] << 8) + payload[0x19]
+
+        if (payload[0x19] == checksum[0] and payload[0x1a] == checksum[1]):
+            pass # success
+        else:
+            print('checksum fail', ['{:02x}'.format(x) for x in checksum])
+
+        if (payload_debug):
+            data['received_payload'] = payload
+
+        return data
+
+    def get_ac_info(self, payload_debug: bool = None) -> dict:
+        """Returns dictionary with A/C info...
+        Not implemented yet, except power state.
+
+        Args:
+            payload_debug (Optional[bool]): add the received payload for debugging
+        """
+        payload = self._send_short_payload(0)
+
+        # first 13 bytes are the same: 22 00 bb 00 07 00 00 00 18 00 01 21 c0
+        data = {}
+        data['state'] = True if (payload[0x0d] & 0b1 == 0b1) else False
+
+        if (payload_debug):
+            data['received_payload'] = payload
+
+        return data
+
+    def set_advanced(self,
+        state: bool = None,
+        mode: str = None,
+        target_temp: float = None,
+        speed: str = None,
+        swing_v: str = None,
+        swing_h: str = None,
+        sleep: bool = None,
+        display: bool = None,
+        health: bool = None,
+        cmnd_0d_rmask: int = 0b100,
+        cmnd_0e_rmask: int = 0b1101,
+        cmnd_18: int = 0b101,
+    ) -> bytes:
+        """Set paramaters of unit and return response.
+        If not all parameters are specificed, will try to derive from the unit's current state.
+        
+        Args:
+            state (bool): power
+            target_temp (float): temperature set point 16<n<32
+            mode (str): cooling, heating, fan, dry, auto
+            speed (str): mute, low, mid, high, turbo (available only for cooling) 
+            swing_h (str): ON, OFF
+            swing_v (str): ON, OFF, 1, 2, 3, 4, 5 (fixed positions)
+            sleep (bool) 
+            display (bool)
+            health (bool)
+            cmnd_0d_rmask (int): unknown
+            cmnd_0e_rmask (int): unknown
+            cmnd_18 (int): unknown
+        """
+        
+        received_state = self.get_state()
```

Comment:
Something tells me that if we understand how this mask works we won't need to call self.get_state() before sending the command. We must do our best to keep communication atomic (we don't want to send more than one request per method).